### PR TITLE
Prevent JS error in pricing page for subdirs sites

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This PR fixes a JS issue that occurs when visiting the pricing page in cloud with a secondary site of a multisite network (with sub directories). See capture below.

### Testing instructions

- Run this PR
- Make sure you have a secondary site available (from a multisite on subdirs configuration). Ping me if you need details about how to set one.
- Open your browser console and visit `http://jetpack.cloud.localhost:3000/pricing/?site=:slug&source=jetpack-plans`. `:slug` should be something like `example.com::other-site`.
- Check that you don't see any error in the console

### Screenshots

Screen Shot 2021-01-13 at 2 50 28 PM

### CODE REVIEW

File: client/components/jetpack/card/jetpack-free-card-button/index.tsx [hunk 1]
```diff
@@ -38,12 +38,17 @@ const JetpackFreeCardButton: FC< JetpackFreeCardButtonProps > = ( {
...
 
 	// If the user is not logged in and there is a site in the URL, we need to construct
 	// the URL to wp-admin from the `site` query parameter
-	const wpAdminUrlFromQuery = urlQueryArgs.site
+	const wpAdminUrlFromQuery = site
 		? getUrlFromParts( {
-				...getUrlParts( `https://${ urlQueryArgs.site }/wp-admin/admin.php` ),
+				...getUrlParts(
+					`https://${
+						isJetpackSecondarySiteSlug( site ) ? site.replace( '::', '/' ) : site
```
Comment (ChaosExAnima) [inline client/components/jetpack/card/jetpack-free-card-button/index.tsx]: Does this code need this function? From what I can tell, it uses regex to check for `::` only to replace it with `/`. Would simply `site.replace( '::', '/' )` work?

**Review by ChaosExAnima — COMMENTED** [review]
Works! Left a question on the code.

File: client/components/jetpack/card/jetpack-free-card-button/index.tsx (hunk 1 above)
Comment (monsieur-z) [reply client/components/jetpack/card/jetpack-free-card-button/index.tsx]: Most likely, but the check seemed safer if `::` was somehow used differently.

Comment (ChaosExAnima) [reply client/components/jetpack/card/jetpack-free-card-button/index.tsx]: Perhaps, but maybe doing a replace and validating via `URL` would be better? I'm concerned there'll be domain names that don't fit a hard-coded regex, such as i18n domain names.

Comment (monsieur-z) [reply client/components/jetpack/card/jetpack-free-card-button/index.tsx]: That's fair. I updated the PR.

**Review by ChaosExAnima — APPROVED** [review]
Awesome, looks great now! Retested and still good as well 👍🏻

**Pull request merged**